src/server/hll_family_test.cc

```diff
@@ -26,6 +29,20 @@ TEST_F(HllFamilyTest, Simple) {
   EXPECT_EQ(CheckedInt({"pfcount", "key"}), 1);
 }
 
+TEST_F(HllFamilyTest, Promote) {
+  int unique_values = 20000;
+  for (int i = 0; i < unique_values; ++i) {
+    std::string newkey = GenerateUniqueValue(i);
+    // sparse hll is promoted to dense at the 1660th+- insertion. But how to verify that?
```

Comment:
How about using the size of the string?
Make sure that it grows until it reaches DENSE_SIZE and then continues in that size until the end of the test.